app/Http/Controllers/Api/AssetsController.php

```diff
@@ -638,6 +648,19 @@ public function update(ImageUploadRequest $request, $id)
             ($request->filled('rtd_location_id')) ?
                 $asset->location_id = $request->get('rtd_location_id') : null;
 
+			/**
+			* rule of set up EOL date and explicit status during update asset
+			*/
+			if ($request->filled('asset_eol_date')) {
```

Comment:
Same as above comment on the `store()` method.

Reply:
For update asset via API (`update()` method), similar as it is for `store()` method.
Only one difference is to check, what is current status of `eol_excplicit` set under asset to make update `asset_eol_date` or not, if purchase date or model is changed.

Reply:
I use this modification by some time (last few months) no issue so far 👍

Reply:
Hey @Robert-Azelis , while this method is working for you, we have lots of people that may be using this feature differently, so need to make sure that any change is going to work for as many people as possible. 

If someone isn't aware of the changes, they might not be aware that they need to send that explicit value now - making their data invalid or misleading. 

If we can do the `diffInMonths()` check than they can continue to just send us `asset_eol_date` and still have a reliable value in `eol_explicit` _and_ we can then also allow them to set that manually if they'd like.

Reply:
hi @spencerrlongg , thank you for your comment, but still don't catch your poin. 
Argument `eol_explicit` is optional used only if user would like to force manually set explicity marker, independents to other arguments (much more usefull for switch on/off `eol_explicit` during update asset - it's not in this part of code. 
The main triger is usage of `asset_eol_date` argument, if has been provided during create or update asset it means user would like to set date manually, not based on automatically calculation. It can be also used in scenario when purchase date is not set, but user would like to keep EOL.

Can you describe scenario or propose exact changes in code (how you would like to use `diffInMonths()` function)?

I'm open for improvements, but need to good understand how it is expected for use :)

Reply:
Hi @spencerrlongg, no feedback from you so far, if proposed changes in **API/AssetsController.php** are so difficult to accept,  we can skip it at this moment, allow to implment other changes and will return to this part under new PR, what do you think about this? 
I don't want to keep on hold this PR.